src/graph.cc

```diff
@@ -660,3 +670,26 @@ void ImplicitDepLoader::CreatePhonyInEdge(Node* node) {
   // this node, it will simply set outputs_ready_ to the correct value.
   phony_edge->outputs_ready_ = true;
 }
+
+bool ImplicitDepNodeLoader::LoadDepfileDepNodes(Edge* edge, std::string path,
+                                                std::vector<Node*>* dep_nodes,
+                                                std::string* err) {
+  dep_nodes_ = dep_nodes;
+  bool ret = LoadDepfileDeps(edge, path, err);
+  dep_nodes_ = NULL;
```

Comment:
What if dep_nodes_ wasn't `NULL` before?

Reply:
Okay that can't happen. But why even set it to `NULL` here?

Reply:
I did this to avoid holding on to a potentially-dangling pointer. This is admittedly a bit dodgy if someone were to call LoadDeps on this loader afterwards. I'm considering removing LoadDepfileDepNodes and adding a direct setter for dep_nodes_, so the calling code would be
```
dep_loader.SetDepNodesOutput(&depfile_deps);
dep_loader.LoadDeps(edge, depfile, &err);
```
instead of the current
```
dep_loader.LoadDepfileDepNodes(edge, depfile, &depfile_deps, &err);
```
Or maybe even just take the vector pointer in the constructor of the custom dep loader to keep things more straightforward since the pointer can then be easily shown as something necessary. That seems the way to go.

Reply:
> Or maybe even just take the vector pointer in the constructor of the custom dep loader to keep things more straightforward since the pointer can then be easily shown as something necessary.

+1

Reply:
PTAL now